connectivity/check/deployment.go

```diff
@@ -650,6 +650,9 @@ func (ct *ConnectivityTest) deleteDeployments(ctx context.Context, client *k8s.C
 		ct.Logf("⌛ [%s] Waiting for namespace %s to disappear", client.ClusterName(), ct.params.TestNamespace)
 		for err == nil {
 			time.Sleep(time.Second)
+			// Retry the namespace deletion in-case the previous delete was
+			// rejected, i.e. by yahoo/k8s-namespace-guard
+			_ = client.DeleteNamespace(ctx, ct.params.TestNamespace, metav1.DeleteOptions{})
```

Comment:
Another thought: it could be possible that the delete never succeeds (e.g. stuck finalizer). Do we want to add a timeout here, so the user doesn't have to Ctl-C?

Reply:
While technically it might, I think that'd be surprising since we're making the namespace in the tests and aren't setting up anything too fancy there. If folks prefer this is setup with a context and times out, I can work on doing that though.